sc-memory/sc-memory/src/sc_action.cpp

```diff
@@ -214,11 +214,27 @@ void ScAction::Finish(ScAddr const & actionStateAddr) noexcept(false)
         "Not able to finish action " << GetActionPrettyString() << GetActionClassPrettyString()
                                      << " because it had already been finished.");
 
-  if (!m_context->IsElement(m_resultAddr))
-    m_resultAddr = m_context->GenerateNode(ScType::ConstNodeStructure);
+  ScIterator5Ptr const resultIt = m_context->CreateIterator5(
```

Comment:
I suspect that this one fails when action has result generated not with this api and `m_resultAddr` wasn't set
in that case connector generation between *this and `m_resultAddr` will throw an exception